### PR TITLE
feat: DamageParticles module

### PR DESCRIPTION
**Pull request by MukjepScarlet**
image

Show health variance of entities

### CODE REVIEW

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleDamageParticles.kt [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+    /**
+     * range = domain = 0.0..1.0
+     */
+    sealed class TransitionChoice(name: String) : Choice(name), (Double) -> Double {
+        override val parent
+            get() = transitionType
+    }
+
+    private object Linear : TransitionChoice("Linear") {
```
Comment (ccetl) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleDamageParticles.kt]: Use the existing "curve" to create a transition function setting.

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleDamageParticles.kt [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    private val transitionY by float("TransitionY", 1.0F, -2.0F..2.0F)
+    private val transitionType = choices(
+        "TransitionType", Linear, arrayOf(
+            Linear, Sine, Exponential, EaseInOut, Cubic, Elastic, Back, Step
+        )
+    )
+
+    private fun Double.doTransition() = transitionType.activeChoice(this)
+
+    private val healthMap = hashMapOf<LivingEntity, Float>()
```
Comment (ccetl) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleDamageParticles.kt]: Perhaps you could use an Object2Float map.

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleDamageParticles.kt (hunk 1 above)
Comment (MukjepScarlet) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleDamageParticles.kt]: Merged.

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleDamageParticles.kt (hunk 2 above)
Comment (MukjepScarlet) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleDamageParticles.kt]: There are several implementations of Object2FloatMap, I choosed one of them without considerations of performance.

**Comment by larryngton2** [comment]
/home/runner/work/LiquidBounce/LiquidBounce/src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleDamageParticles.kt:76:17: Missing braces on this branch, add them. [BracesOnIfStatements]

**Review by ccetl — COMMENTED** [review]
You're sure the health map doesn't have to be cleared when you rejoin a server?

Looks good otherwise.

**Comment by MukjepScarlet** [comment]
It will remove all entities not existing or already dead from health map on every tick. We can add another check if player quit from a game.

**Review by 1zun4 — APPROVED** [review]

**Pull request merged**